File: frontend-master/src/app/features/hotels/models/hotel.model.ts
import { LocationModel } from './location.model';

export interface HotelModel {
  id: number;
  name: string;
  createdAt: string;
  location: LocationModel;
  // TODO ???
  amenities: string[];
  description: string;
}
export interface HotelDetailModel extends HotelModel{
  amenities: string[];
  description: string;
  creatorName: string;
  creatorId:number;
}

File: backend-master/bd/src/main/java/ru/relex/miniBooking/bd/model/HotelModel.java
package ru.relex.miniBooking.bd.model;

import lombok.*;

import java.util.Date;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class    HotelModel {
    long id;
    String name;
    String country;
    String city;
    String street;
    String house;
    String description;
    String[] amenities;
    Date createdAt;
}
